File: SDKs/twilio.com/twilio_voice_v1/1.36.0/python/src/sdk/models/operations/updateconnectionpolicytarget.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared
UPDATE_CONNECTION_POLICY_TARGET_SERVERS = [
	"https://voice.twilio.com",
]


@dataclass
class UpdateConnectionPolicyTargetPathParams:
    connection_policy_sid: str = field(default=None, metadata={'path_param': { 'field_name': 'ConnectionPolicySid', 'style': 'simple', 'explode': False }})
    sid: str = field(default=None, metadata={'path_param': { 'field_name': 'Sid', 'style': 'simple', 'explode': False }})
    

@dataclass
class UpdateConnectionPolicyTargetUpdateConnectionPolicyTargetRequest:
    enabled: Optional[bool] = field(default=None, metadata={'form': { 'field_name': 'Enabled' }})
    friendly_name: Optional[str] = field(default=None, metadata={'form': { 'field_name': 'FriendlyName' }})
    priority: Optional[int] = field(default=None, metadata={'form': { 'field_name': 'Priority' }})
    target: Optional[str] = field(default=None, metadata={'form': { 'field_name': 'Target' }})
    weight: Optional[int] = field(default=None, metadata={'form': { 'field_name': 'Weight' }})
    

@dataclass
class UpdateConnectionPolicyTargetSecurity:
    account_sid_auth_token: shared.SchemeAccountSidAuthToken = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class UpdateConnectionPolicyTargetRequest:
    server_url: Optional[str] = field(default=None)
    path_params: UpdateConnectionPolicyTargetPathParams = field(default=None)
    request: Optional[UpdateConnectionPolicyTargetUpdateConnectionPolicyTargetRequest] = field(default=None, metadata={'request': { 'media_type': 'application/x-www-form-urlencoded' }})
    security: UpdateConnectionPolicyTargetSecurity = field(default=None)
    

@dataclass
class UpdateConnectionPolicyTargetResponse:
    content_type: str = field(default=None)
    status_code: int = field(default=None)
    voice_v1_connection_policy_connection_policy_target: Optional[shared.VoiceV1ConnectionPolicyConnectionPolicyTarget] = field(default=None)
    

File: SDKs/twilio.com/twilio_voice_v1/1.36.0/typescript/src/sdk/models/operations/updateconnectionpolicytarget.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const UPDATECONNECTIONPOLICYTARGET_SERVERS = [
	"https://voice.twilio.com",
];



export class UpdateConnectionPolicyTargetPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=ConnectionPolicySid" })
  connectionPolicySid: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=Sid" })
  sid: string;
}


export class UpdateConnectionPolicyTargetUpdateConnectionPolicyTargetRequest extends SpeakeasyBase {
  @Metadata({ data: "form, name=Enabled;" })
  enabled?: boolean;

  @Metadata({ data: "form, name=FriendlyName;" })
  friendlyName?: string;

  @Metadata({ data: "form, name=Priority;" })
  priority?: number;

  @Metadata({ data: "form, name=Target;" })
  target?: string;

  @Metadata({ data: "form, name=Weight;" })
  weight?: number;
}


export class UpdateConnectionPolicyTargetSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class UpdateConnectionPolicyTargetRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  pathParams: UpdateConnectionPolicyTargetPathParams;

  @Metadata({ data: "request, media_type=application/x-www-form-urlencoded" })
  request?: UpdateConnectionPolicyTargetUpdateConnectionPolicyTargetRequest;

  @Metadata()
  security: UpdateConnectionPolicyTargetSecurity;
}


export class UpdateConnectionPolicyTargetResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  voiceV1ConnectionPolicyConnectionPolicyTarget?: shared.VoiceV1ConnectionPolicyConnectionPolicyTarget;
}
